components/automate-ui/src/app/modules/user/user-details/user-details.component.ts

```diff
@@ -54,8 +55,8 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
       pluck('id'),
       filter(identity),
       takeUntil(this.isDestroyed))
-      .subscribe((id: string) => {
-        this.store.dispatch(new GetUser({ id }));
+      this.route.params.subscribe(params => {
+        this.store.dispatch(new GetUser({ id: params.id }));
```

Comment:
It looks like you are trying to nest a second `subscribe` inside the first `subscribe`, which does not quite make sense. But I don't believe the code as written is doing what you think. You actually have two separate subscriptions--the first one stops at L57 and does nothing (you will see the linter complaining about a missing semicolon there). The second one starts at L58 and does what the code did originally, but it is missing the vital `takeUntil`.

There are two possibilities for this block. Using the approach you were suggesting would look like this:
```
    this.route.params.pipe(
      takeUntil(this.isDestroyed))
      .subscribe(params => {
        this.store.dispatch(new GetUser({ id: params.id }));
      });
```
Or, the approach I previously suggested would also work:
```
    this.store.pipe(
      select(routeParams),
      pluck('id'),
      filter(identity),
      first())
      .subscribe((id: string) => {
          this.store.dispatch(new GetUser({ id }));
      });
```
As the first is a bit shorter, probably makes sense to use that one.

Reply:
Thanks for the suggestions, I have updated the changes accordingly, please have a look :)

Reply:
Hmm... because of the failing tests, we might have to go with the second version, which I think should allow the tests to pass. Try that and see if it works.

Reply:
yes, It works, Thanks :)